components/director/internal/repo/common.go

```diff
@@ -11,32 +11,44 @@ type Condition struct {
 	Val   string
 }
 
-func getAllArgs(tenant string, conditions Conditions) []interface{} {
-	allArgs := []interface{}{tenant}
+func getAllArgs(tenant *string, conditions Conditions) []interface{} {
+	allArgs := []interface{}{}
+	if tenant != nil {
+		allArgs = append(allArgs, tenant)
+	}
 	for _, idAndVal := range conditions {
 		allArgs = append(allArgs, idAndVal.Val)
 	}
 	return allArgs
 }
 
-func appendEnumeratedConditions(query string, startIdx int, conditions Conditions) string {
-	out := query
+func appendEnumeratedConditions(builder *strings.Builder, startIdx int, conditions Conditions) *strings.Builder {
```

Comment:
Agree, it might be confusing. I had to pass the pointer to builder to keep using it, but I don't really need to return it. I will remove the returned value and rename this function so it's behaviour won't be mistaken with how Go's `append` works.